packages/tc-schema-validator/validation/ad-hoc/relationship-consistency.js

```diff
@@ -90,12 +90,15 @@ const validateRelationshipConsistency = () => {
 			);
 
 			assert.equal(
-				fromPropDefs.length,
+				fromPropDefs.filter(
+					({ deprecationReason }) => !deprecationReason,
```

Comment:
Highlight #11 These changes allow for deprecating a relationship under one name and adding a new one that uses the same underlying data. The different test schema prompted this, but it is something that feels necessary: The tests should expect that active relationship properties come in pairs. but should be lenient on deprecated ones